src/pages/Pokemon/Stats.js

```diff
@@ -1,26 +1,146 @@
 import React from 'react';
 import './pokemon.css'
 
-
 const Stats = ({ pokemonItem }) => {
+
   let statsArray = pokemonItem.stats;
-    return(
-        <div className='stats-wrapper'>
-        <div className="statsDiv">
-            <h3 className='stats-header'>Stats</h3>
-        {statsArray.map((item) => {
-          return (
-            <div
-              className={`statsCard ${item.stat.name}`}
-              key={item.stat.name}>
-              {item.stat.name.charAt(0).toUpperCase() +
-                item.stat.name.slice(1)}
-            </div>
-          );
-        })}
-      </div>
+  // console.log(statsArray)
+
+  let numsArray = []
+  for (let i = 0; i < statsArray.length; i++) {
+    const newObj = statsArray[i]
+    // console.log(newObj)
+
+    const nums = (newObj.base_stat)
+    // console.log(nums)
+    numsArray.push(Math.round(nums / 17))
+  }
+  // console.log(numsArray)
+
+  // const ulList = document.querySelectorAll(".stat-ul-child")
+  // console.log(ulList)
+
+  // const ulArray = Array.from(ulList)
+  // console.log(ulArray)
+
+  // let liArray = []
+  // for(let i = 0; i < ulArray.length; i++){
+  //   liArray.push(Array.from(ulArray[i].children))
+  // }
+  // console.log(liArray)
+
+  // let liArr = []
+  // for(let j = 0; j < liArray.length; j++){
+  //   liArr.push((liArray[j]))
+  // }
+  // console.log(liArr)
+
+  // for(let j = 0; j < ulArray.length; j++)
+  // let liArray1 = liArray[j]
+  // console.log(liArray1)
+
+  // Original for loop example that I came up with:
+
+  // let liArray1 = Array.from(ulArray[0].children)
+  // // console.log(liArray1.length)
+  // for (let i = 0, j = 14; i < liArray1.length && j >= 0; i++, j--) {
+  //   i < numsArray[0] ? liArray1[j].style.backgroundColor = ("#30a7d7") : null
+  // }
+
+  // With j removed:
+
+  // let liArray1 = Array.from(ulArray[0].children)
+  // for (let i = 0; i < liArray1.length; i++) {
+  //   i < numsArray[0] ? liArray1[liArray1.length - (i + 1)].style.backgroundColor = ("#30a7d7") : null
+  // }
+
+  // With j removed and column reverse enabled:
+
+  // let liArray1 = Array.from(ulArray[0].children)
+  // for (let i = 0; i < liArray1.length; i++) {
+  //   i < numsArray[0] ? liArray1[i].style.backgroundColor = ("#30a7d7") : null
+  // }
+
+  // forEach version:
+
+  // let liArray1 = Array.from(ulArray[0].children)
+  // liArray1.forEach((li, i) =>
+  //   i < numsArray[0] ? (liArray1[i]).style.backgroundColor = ("#30a7d7") : null
+  // )
+
+  // Map version:
+
+  // console.log(ulArray[0].children)
+  // let liArray1 = Array.from(ulArray[0].children)
+  // console.log(liArray1)
+  // let lis = 15
+  // let lisArray = Array.from(new Array(lis))
+  // console.log(lisArray)
+
+  const StatBar = (props) => {
```

Comment:
we should explicitly define the props here mate, rather than just getting the whole object

so it should be:

`const StatBar = ({ numOfBarsColored, statName }) => { }`